skimage/measure/fit.py

```diff
@@ -782,15 +785,18 @@ def ransac(data, model_class, min_samples, residual_threshold,
     best_inliers = None
 
     random_state = check_random_state(random_state)
-
+    if isinstance(min_samples, float):
+        if not (0 < min_samples <= 1):
+            raise ValueError("`min_samples` as ration must be in range (0, 1)")
+        min_samples = int(min_samples * len(data))
     if min_samples < 0:
         raise ValueError("`min_samples` must be greater than zero")
 
     if max_trials < 0:
         raise ValueError("`max_trials` must be greater than zero")
 
     if stop_probability < 0 or stop_probability > 1:
```

Comment:
I would change this to `if not 0 < stop_probability < 1:` to match the error text. (Currently, `stop_probability=0` will not be caught here!)